tiledb/sm/array_schema/array_schema.cc

```diff
@@ -255,14 +253,15 @@ const FilterPipeline* ArraySchema::filters(const std::string& name) const {
   // Dimension (if filters not set, return default coordinate filters)
   auto dim_it = dim_map_.find(name);
   assert(dim_it != dim_map_.end());
-  auto ret = dim_it->second->filters();
-  return (ret != nullptr) ? ret : coords_filters();
+  const auto& ret = dim_it->second->filters();
+  return !ret.empty() ? ret : coords_filters();
 }
 
-const FilterPipeline* ArraySchema::coords_filters() const {
-  return &coords_filters_;
+const FilterPipeline& ArraySchema::coords_filters() const {
+  return coords_filters_;
 }
 
+/*
```

Comment:
remove commented-out code block if not needed

Reply:
Ooops missed that. I removed it in a new upcoming PR.